src/Simulation/Simulators/SparseSimulator/Native/factory.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// Manages simulators in a vector of pointers to simulators
+
+#include "factory.hpp"
+#include "SparseSimulator.h"
+#include "types.h"
+#include <iostream>
+#include <shared_mutex>
+
+
+namespace Microsoft
+{
+namespace Quantum
+{
+namespace SPARSESIMULATOR
+{
+// Ensures exclusive access to _simulators, the vector of simulators
+std::shared_mutex _mutex;
+std::vector<std::shared_ptr<SparseSimulator>> _simulators;
+
+unsigned createSimulator(logical_qubit_id num_qubits)
+{
+    if (num_qubits > MAX_QUBITS)
+        throw std::runtime_error("Max number of qubits is 1024!");
```

Comment:
Probably not a good idea to include 1024 in the message when MAX_QUBITS is defined elsewhere. Whoever would change MAX_QUBITS will likely forget to update the message.

Reply:
Removed 1024 from message here: https://github.com/sam-jaques/qsharp-runtime/pull/1